packages/amplify-e2e-core/src/index.ts

```diff
@@ -31,6 +34,14 @@ export function isCI(): boolean {
   return process.env.CI && process.env.CIRCLECI ? true : false;
 }
 
+export function injectSessionToken() {
+  const credentialsContents = ini.parse(fs.readFileSync(pathManager.getAWSCredentialsFilePath()).toString());
+  for (const key of Object.keys(credentialsContents)) {
+    credentialsContents[key].aws_session_token = process.env.AWS_SESSION_TOKEN;
+  }
+  fs.writeFileSync(pathManager.getAWSCredentialsFilePath(), ini.stringify(credentialsContents));
+}
+
```

Comment:
I would prefer if this happened as part of setting up the profile after running the configure command to ensure the profile gets set correctly and done only once
https://github.com/aws-amplify/amplify-cli/blob/cd5d33aa822ceeb19a1af847d8c3eab0f1d10632/packages/amplify-e2e-tests/src/configure_tests.ts#L5-L17